keps/sig-cloud-provider/003-testgrid-e2e.md

```diff
@@ -0,0 +1,224 @@
+
+First install sonobuoy. The following command adds it to the `$GOBIN` environment variable which is expected to be part of your `$PATH` environment variable.
+```
+$ go get -u github.com/heptio/sonobuoy
+```
+
+#### Running Conformance Tests with Sonobuoy
+
+You can then start e2e tests by simply running:
+```
+$ sonobuoy run
+Running plugins: e2e, systemd-logs
+INFO[0001] created object                                name=heptio-sonobuoy namespace= resource=namespaces
+INFO[0001] created object                                name=sonobuoy-serviceaccount namespace=heptio-sonobuoy resource=serviceaccounts
+INFO[0001] created object                                name=sonobuoy-serviceaccount-heptio-sonobuoy namespace= resource=clusterrolebindings
+INFO[0001] created object                                name=sonobuoy-serviceaccount namespace= resource=clusterroles
+INFO[0001] created object                                name=sonobuoy-config-cm namespace=heptio-sonobuoy resource=configmaps
+INFO[0002] created object                                name=sonobuoy-plugins-cm namespace=heptio-sonobuoy resource=configmaps
+INFO[0002] created object                                name=sonobuoy namespace=heptio-sonobuoy resource=pods
+INFO[0002] created object                                name=sonobuoy-master namespace=heptio-sonobuoy resource=services
+```
+
+You can then check the status of your e2e tests by running
+```
+$ sonobuoy status
+PLUGIN		STATUS	COUNT
+e2e		running	1
+systemd_logs	running	3
+
+Sonobuoy is still running. Runs can take up to 60 minutes.
+```
+
+E2E tests can take up to an hour. Once your tests are done, you can download a snapshot of your results like so:
+```
+$ sonobuoy retrieve .
+```
+
+Once you have the following snapshot, extract it's contents like so
+```
+$ mkdir ./results; tar xzf *.tar.gz -C ./results
+```
+
+At this point you should have the log file and JUnit results from your tests:
+```
+results/plugins/e2e/results/e2e.log
+results/plugins/e2e/results/junit_01.xml
+```
+
+#### How to Upload Conformance Test Results to Testgrid
+
+##### Requesting a GCS Bucket
+
+Testgrid requires that you store results in a publicly readable GCS bucket. If for whatever reason you cannot set up a GCS bucket, please contact @BenTheElder or more generally the [gke-kubernetes-engprod](mailto:gke-kubernetes-engprod@google.com) team to arrange for a Google [GKE](https://cloud.google.com/kubernetes-engine/) EngProd provided / maintained bucket for hosting your results.
+
+##### Authenticating to your Testgrid Bucket
+
+Assuming that you have a publicly readable bucket provided by the GKE team, you should have been provided a service account JSON file which you can use with [gcloud](https://cloud.google.com/sdk/downloads) to authenticate with your GCS bucket.
+
+```
+$ gcloud auth activate-service-account --key-file /path/to/k8s-conformance-serivce-accout.json
```

Comment:
oh.. that's quite handy. Is that the preferred method going forward?

Reply:
either works fine, but I think for some setups this will be easier? It also means the cloud providers just need to have the gcloud tools available, they don't necessarily need to be aware of how to use them